### ISSUE TITLE
fail2ban issue: stderr: 'iptables: No chain/target/match by that name.'

### ISSUE THREAD

**Issue by ktbgh** [docker-mailserver/docker-mailserver#2141]
### Subject

ISSUE: fail2ban - stderr: 'iptables: No chain/target/match by that name in fail2ban logfile

### Description

I have docker-mailserver running longer time and it runs very well. The only thing what not works was fail2ban.
fail2ban is running and the NET_ADMIN in the cap_add section in yml files was set.

### logged errors

here is a output of `/var/log/fail2ban.log` logfile  of the docker container when I try to login with a wrong password:

```
2021-08-21 13:06:06,440 fail2ban.utils          [8728]: ERROR   7f5910820030 -- stderr: 'iptables: No chain/target/match by that name.'
2021-08-21 13:06:06,440 fail2ban.utils          [8728]: ERROR   7f5910820030 -- returned 1
2021-08-21 13:06:06,441 fail2ban.actions        [8728]: ERROR   Failed to execute ban jail 'dovecot' action 'iptables-multiport' info 'ActionInfo({'ip': 'xx.xx.xx.xx', 'family': 'inet4', 'ip-rev': 'xx.xx.xx.xx.', 'ip-host': 'ppp-xx-xx-xx-xx.home.otenet.gr', 'fid': 'xx.xx.xx.xx', 'failures': 1, 'time': 1629543966.2305715, 'matches': 'Aug 21 12:46:35 myworld dovecot: imap-login: Aborted login (auth failed, 1 attempts in 3 secs): user=<xx.xx@bla.de>, method=PLAIN, rip=xx.xx.xx.xx, lip=yy.yy.yy.yy, TLS, session=<lwFSfw/K96xeQjnr>', 'restored': 1, 'F-*': {'matches': [['', 'Aug 21 12:46:35', ' myworld dovecot: imap-login: Aborted login (auth failed, 1 attempts in 3 secs): user=<xx.xx@bla.de>, method=PLAIN, rip=xx.xx.xx.xx, lip=yy.yy.yy.yy, TLS, session=<lwFSfw/K96xeQjnr>']], 'failures': 1, 'ip4': 'xx.xx.xx.xx'}, 'ipmatches': 'Aug 21 12:46:35 myworld dovecot: imap-login: Aborted login (auth failed, 1 attempts in 3 secs): user=<xx.xx@bla.de>, method=PLAIN, rip=xx.xx.xx.xx, lip=yy.yy.yy.yy, TLS, session=<lwFSfw/K96xeQjnr>\nAug 21 12:46:41 myworld postfix/submission/smtpd[2843]: warning: ppp-94-66-57-235.home.otenet.gr[xx.xx.xx.xx]: SASL PLAIN authentication failed: \nAug 21 12:46:51 myworld dovecot: imap-login: Aborted login (auth failed, 1 attempts in 6 secs): user=<xx.xx@bla.de>, method=PLAIN, rip=xx.xx.xx.xx, lip=yy.yy.yy.yy, TLS, session=<LMkMgA/KyxNeQjnr>\nAug 21 12:46:54 myworld postfix/submission/smtpd[2843]: warning: ppp-94-66-57-235.home.otenet.gr[xx.xx.xx.xx]: SASL PLAIN authentication failed: \nAug 21 12:46:59 myworld dovecot: imap-login: Aborted login (auth failed, 1 attempts in 2 secs): user=<xx.xx@bla.de>, method=PLAIN, rip=xx.xx.xx.xx, lip=yy.yy.yy.yy, TLS, session=<YgDCgA/K501eQjnr>\nAug 21 12:47:02 myworld postfix/submission/smtpd[2843]: warning: ppp-94-66-57-235.home.otenet.gr[xx.xx.xx.xx]: SASL PLAIN authentication failed: ', 'ipjailmatches': 'Aug 21 12:46:35 myworld dovecot: imap-login: Aborted login (auth failed, 1 attempts in 3 secs): user=<xx.xx@bla.de>, method=PLAIN, rip=xx.xx.xx.xx, lip=yy.yy.yy.yy, TLS, session=<lwFSfw/K96xeQjnr>\nAug 21 12:46:51 myworld dovecot: imap-login: Aborted login (auth failed, 1 attempts in 6 secs): user=<xx.xx@bla.de>, method=PLAIN, rip=xx.xx.xx.xx, lip=yy.yy.yy.yy, TLS, session=<LMkMgA/KyxNeQjnr>\nAug 21 12:46:59 myworld dovecot: imap-login: Aborted login (auth failed, 1 attempts in 2 secs): user=<xx.xx@bla.de>, method=PLAIN, rip=xx.xx.xx.xx, lip=yy.yy.yy.yy, TLS, session=<YgDCgA/K501eQjnr>', 'ipfailures': 6, 'ipjailfailures': 3})': Error starting action Jail('dovecot')/iptables-multiport
```

**docker-compose.yml**

```
version: '3.8'
services:
  mail:
    image: docker.io/mailserver/docker-mailserver:latest
    hostname: ${HOSTNAME}
    domainname: ${DOMAINNAME}
    #    container_name: ${CONTAINER_NAME}
    container_name: myworld_mail
    ports:
      - "4190:4190"
      - "5525:25"
      - "55143:143"
      - "55587:587"
      - "55993:993"
      - "55465:465"
      - "55995:995"
    volumes:
      - maildata:/var/mail
      - mailstate:/var/mail-state
      - maillogs:/var/log/mail
      - ./config/:/tmp/docker-mailserver/
      - "./timezone:/etc/timezone:ro"
      - "/etc/localtime:/etc/localtime:ro"
        #      - "./hosts:/etc/hosts:rw"
      - "./resolv.conf:/etc/resolv.conf:ro"
      - /etc/letsencrypt/live/mail.xxx.domainname:/tls:ro
    env_file:
      - .env
      - mailserver.env
    restart: always
    stop_grace_period: 1m
    cap_add:
      - NET_ADMIN
      - SYS_PTRACE
  roundcube:
    image: robbertkl/roundcube
    container_name: roundcube
    restart: always
    depends_on:
      - mail
    volumes:
      - ./config/rcm_config.inc.php:/var/www/config/config.inc.php
      - ./config/rcm_managesieve.config.inc.php:/var/www/plugins/managesieve/config.inc.php
      - rcm_data:/var/www
      - "./timezone:/etc/timezone:ro"
      - "/etc/localtime:/etc/localtime:ro"
    environment:
      - ROUNDCUBE_USER_FILE="/var/www/config/mailboxpwfile"
    ports:
      - 80:80
volumes:
  maildata:
    driver: local
  mailstate:
    driver: local
  maillogs:
    driver: local
  rcm_data:
    driver: local
```
**mailserver DEBUG Start output**

[mailserver-start.debug-output.txt](https://github.com/docker-mailserver/docker-mailserver/files/7026970/mailserver-start.debug-output.txt)

I cannot say how to reproduce this issue. 
Please let me know if additional information is required.
Thank you very much!

**Comment by georglauterbach** [comment]
This issue does not follow [our official issue guidelines](https://docker-mailserver.github.io/docker-mailserver/edge/contributing/issues-and-pull-requests/) and will therefore not be worked on until this is resolved.

**Comment by ktbgh** [comment]
I'm very sorry. 
thank you for your hint. 
I'm a beginner here and I'm not sure i filled correctly the issue now.
I have edited the Issue now.

**Comment by ktbgh** [comment]
I'm very sorry. 
thank you for your hint. 
I'm a beginner here and I'm not sure i filled correctly the issue now.
I have edited the Issue now.

**Comment by georglauterbach** [comment]
@casperklein any idea?

@ktbgh I would start by adding the `NET_RAW` capability and see if that helps.

**Comment by casperklein** [comment]
What OS is your host running?

What is the output of:

 - `docker exec -it mail iptables -L -n`
 - `iptables -L -n`

**Comment by ktbgh** [comment]
with 

```
    cap_add:
      - NET_ADMIN
      - NET_RAW
      - SYS_PTRACE
```

in docker-compose.yml file I have the same error message when I provoke a wrong password login:

```
root@myworld:/# tail -f /var/log/fail2ban.log
2021-08-22 13:14:07,901 fail2ban.utils          [491]: ERROR   7f613fdad030 -- stderr: 'iptables: No chain/target/match by that name.'
2021-08-22 13:14:07,901 fail2ban.utils          [491]: ERROR   7f613fdad030 -- returned 1
2021-08-22 13:14:07,901 fail2ban.actions        [491]: ERROR   Failed to execute ban jail 'dovecot' action 'iptables-multiport' info 'ActionInfo({'ip': '94.66.57.235', 'family': 'inet4', 'ip-rev': '235.57.66.94.', 'ip-host': 'ppp-94-66-57-235.home.otenet.gr', 'fid': '94.66.57.235', 'failures': 1, 'time': 1629630847.0, 'matches': 'Aug 22 13:14:07 myworld dovecot: imap-login: Aborted login (auth failed, 1 attempts in 2 secs): user=<xxxxx@xxx.domainname>, method=PLAIN, rip=94.66.57.235, lip=192.168.208.2, TLS, session=<UYyl/yPKcEdeQjnr>', 'restored': 0, 'F-*': {'matches': [('', 'Aug 22 13:14:07', ' myworld dovecot: imap-login: Aborted login (auth failed, 1 attempts in 2 secs): user=<xxxxx@xxx.domainname>, method=PLAIN, rip=94.66.57.235, lip=192.168.208.2, TLS, session=<UYyl/yPKcEdeQjnr>')], 'failures': 1, 'ip4': '94.66.57.235'}, 'ipmatches': 'Aug 22 13:14:07 myworld dovecot: imap-login: Aborted login (auth failed, 1 attempts in 2 secs): user=<xxxxx@xxx.domainname>, method=PLAIN, rip=94.66.57.235, lip=192.168.208.2, TLS, session=<UYyl/yPKcEdeQjnr>', 'ipjailmatches': 'Aug 22 13:14:07 myworld dovecot: imap-login: Aborted login (auth failed, 1 attempts in 2 secs): user=<xxxxx@xxx.domainname>, method=PLAIN, rip=94.66.57.235, lip=192.168.208.2, TLS, session=<UYyl/yPKcEdeQjnr>', 'ipfailures': 1, 'ipjailfailures': 1})': Error starting action Jail('dovecot')/iptables-multiport
2021-08-22 13:14:15,796 fail2ban.utils          [491]: ERROR   7f613fd943f0 -- stderr: 'iptables: No chain/target/match by that name.'
2021-08-22 13:14:15,796 fail2ban.utils          [491]: ERROR   7f613fd943f0 -- returned 1
2021-08-22 13:14:15,796 fail2ban.actions        [491]: ERROR   Failed to execute ban jail 'postfix-sasl' action 'iptables-multiport' info 'ActionInfo({'ip': '94.66.57.235', 'family': 'inet4', 'ip-rev': '235.57.66.94.', 'ip-host': 'ppp-94-66-57-235.home.otenet.gr', 'fid': '94.66.57.235', 'failures': 1, 'time': 1629630855.0, 'matches': 'Aug 22 13:14:15 myworld postfix/submission/smtpd[1613]: warning: ppp-94-66-57-235.home.otenet.gr[94.66.57.235]: SASL PLAIN authentication failed: ', 'restored': 0, 'F-*': {'matches': [('', 'Aug 22 13:14:15', ' myworld postfix/submission/smtpd[1613]: warning: ppp-94-66-57-235.home.otenet.gr[94.66.57.235]: SASL PLAIN authentication failed: ')], 'failures': 1, 'ip4': '94.66.57.235'}, 'ipmatches': 'Aug 22 13:14:07 myworld dovecot: imap-login: Aborted login (auth failed, 1 attempts in 2 secs): user=<xxxxx@xxx.domainname>, method=PLAIN, rip=94.66.57.235, lip=192.168.208.2, TLS, session=<UYyl/yPKcEdeQjnr>\nAug 22 13:14:15 myworld postfix/submission/smtpd[1613]: warning: ppp-94-66-57-235.home.otenet.gr[94.66.57.235]: SASL PLAIN authentication failed: ', 'ipjailmatches': 'Aug 22 13:14:15 myworld postfix/submission/smtpd[1613]: warning: ppp-94-66-57-235.home.otenet.gr[94.66.57.235]: SASL PLAIN authentication failed: ', 'ipfailures': 2, 'ipjailfailures': 1})': Error starting action Jail('postfix-sasl')/iptables-multiport

```

**Comment by ktbgh** [comment]
```
[root@myworld ~]# cat /etc/centos-release
CentOS Linux release 8.4.2105
```

```
[root@myworld ~]# docker exec -it myworld_mail iptables -L -n
Chain INPUT (policy ACCEPT)
target     prot opt source               destination

Chain FORWARD (policy ACCEPT)
target     prot opt source               destination

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination

Chain f2b-dovecot (0 references)
target     prot opt source               destination
RETURN     all  --  0.0.0.0/0            0.0.0.0/0

Chain f2b-postfix-sasl (0 references)
target     prot opt source               destination
RETURN     all  --  0.0.0.0/0            0.0.0.0/0

[host-iptables-L.txt](https://github.com/docker-mailserver/docker-mailserver/files/7027304/host-iptables-L.txt)


```

**Comment by casperklein** [comment]
> I have docker-mailserver running longer time

And probably without updates 😉

You are not running the _latest_ image of DMS. You can verify this by executing: `docker exec -it mail cat /VERSION`. This should return "10.1.1" (when you are using the latest version).
You'll probably see `No such file or directory`.

To update use:

```
docker-compose pull
docker-compose down
docker-compose up -d mail
```

Make sure, that you read the [CHANGELOG](https://github.com/docker-mailserver/docker-mailserver/blob/master/CHANGELOG.md) before.

PS: It's best practice, not to use the `latest` tag, but a specific version tag in `docker-compose.yml`. That way you ensure, that you won't be hit by breaking change, after a restart or so.

**Comment by casperklein** [comment]
After the update, this

`<mailserver-name> | [[  INF  ]]  Welcome to docker-mailserver` (from your `mailserver-start.debug-output.txt`)

should change to

`<mailserver-name> | [ TASKLOG ]  Welcome to docker-mailserver 10.1.0`

**Comment by ktbgh** [comment]
mmhm you're right .... I updated sometimes but the latest update is missed. 
Let me implement and test it. I will come back soon.
Thank you.

**Comment by ktbgh** [comment]
Okay, i have updated now to the latest 10.1 release - the error occurs again. BUT - I found a fail2ban-jail.cf file in config folder, with this I have tested on this issue some months ago without success and after stop testing, I forgot to delete this file.
This was the file which has produced the errors. 
after deleting it now, fail2ban works as it should. 
As many times, here too are sitting the problem in front of the computer. :-/
Thank you for your support, I close now this issue.
Kind regards
Karl
